### PR TITLE
core: entity locks

### PR DESCRIPTION
**Pull request by spaced4ndy**

### CODE REVIEW

File: tests/ChatTests/Profiles.hs [hunk 1]
```diff
@@ -179,10 +179,12 @@ testMultiWordProfileNames =
...
-      alice <## "member #'Our Team' 'Cath Johnson' does not have direct connection, creating"
-      alice <## "contact for member #'Our Team' 'Cath Johnson' is created"
-      alice <## "sent invitation to connect directly to member #'Our Team' 'Cath Johnson'"
-      alice <# "@'Cath Johnson' hello"
+      alice
+        <### [ "member #'Our Team' 'Cath Johnson' does not have direct connection, creating",
+               "contact for member #'Our Team' 'Cath Johnson' is created",
+               "sent invitation to connect directly to member #'Our Team' 'Cath Johnson'",
+               WithTime "@'Cath Johnson' hello"
+             ]
```
Comment (spaced4ndy) [inline tests/ChatTests/Profiles.hs:187]: this fix is unrelated to locks - in similar tests we also expect in any order

File: tests/ChatTests/Files.hs [hunk 2]
```diff
@@ -787,7 +787,8 @@ testXFTPCancelRcvRepeat =
       bob ##> "/fr 1 ./tests/tmp"
       bob
         <### [ "saving file 1 from alice to ./tests/tmp/testfile_1",
-               "started receiving file 1 (testfile) from alice"
+               "started receiving file 1 (testfile) from alice",
+               StartsWith "chat db error: SERcvFileNotFoundXFTP"
```
Comment (spaced4ndy) [inline tests/ChatTests/Files.hs:791]: this test always had error but a different one - CENoRcvFileUser, it was muted by log level. Error happens because agent reports next downloaded chunk when cancel happened in progress

File: src/Simplex/Chat/Store/Connections.hs [hunk 3]
```diff
@@ -37,6 +39,49 @@ import qualified Simplex.Messaging.Agent.Store.SQLite.DB as DB
 import Simplex.Messaging.Crypto.Ratchet (PQSupport)
 import Simplex.Messaging.Util (eitherToMaybe)
 
+getConnectionEntityId :: DB.Connection -> AgentConnId -> ExceptT StoreError IO ConnectionEntityId
```
Comment (spaced4ndy) [inline src/Simplex/Chat/Store/Connections.hs:42]: couldn't split getConnectionEntity, because its first step is reading connection by id, and it also could be updated so it should be read after taking lock. also it's used in other places as well

File: src/Simplex/Chat/Protocol.hs [hunk 4]
```diff
@@ -112,6 +113,14 @@ agentToChatVersion v
   | v < pqdrSMPAgentVersion = initialChatVersion
   | otherwise = pqEncryptionCompressionVersion
 
+data ConnectionEntityId
```
Comment (spaced4ndy) [inline src/Simplex/Chat/Protocol.hs:116]: probably would make more sense to have this together with ConnectionEntity somewhere in Types, but since ConnectionEntity is currently here, didn't move

**Review by dpwiz — DISMISSED** [review]
